tests/virt_autotest/reboot_and_wait_up_normal.pm

```diff
@@ -14,11 +14,27 @@ use strict;
 use warnings;
 use testapi;
 use base "reboot_and_wait_up";
+use Utils::Backends 'is_remote_backend';
+use ipmi_backend_utils;
 
 sub run {
     my $self    = shift;
     my $timeout = 180;
-    $self->reboot_and_wait_up($timeout);
+
+    #get the version that the host is installed to
+    my $host_installed_version = get_var('VERSION_TO_INSTALL', get_var('VERSION', ''));    #format 15 or 15-SP1
```

Comment:
you repeating this 4 lines 3 times , maybe it is better to extract it to a function ? 

something like this :

```
sub is_installed_equal_upgrade() {
my $host_installed_version = get_var('VERSION_TO_INSTALL', get_var('VERSION', '')); 
$host_installed_version) = $host_installed_version =~ /^(\d+)/;
    #get the version that the host should upgrade to
    my $host_upgrade_version = get_var('UPGRADE_PRODUCT', 'sles-1-sp0');                   #format sles-15-sp0
    ($host_upgrade_version) = $host_upgrade_version =~ /sles-(\d+)-sp/i;
return "$host_installed_version" eq "$host_upgrade_version";
}
```
my $host_installed_version = get_var('VERSION_TO_INSTALL', get_var('VERSION', ''));

Reply:
The same problem as the above. I can make the change if you are very sure, namely get('VERSION') always work.

Reply:
looks like I mess up something when I was writing this comment sorry . actually it is not related to `VERSION` variable at all . it is about suggestion to extract duplicate code a function : 

```
sub is_installed_equal_upgrade() {
my $host_installed_version = get_var('VERSION_TO_INSTALL', get_var('VERSION', '')); 
$host_installed_version) = $host_installed_version =~ /^(\d+)/;
    #get the version that the host should upgrade to
    my $host_upgrade_version = get_var('UPGRADE_PRODUCT', 'sles-1-sp0');                   #format sles-15-sp0
    ($host_upgrade_version) = $host_upgrade_version =~ /sles-(\d+)-sp/i;
return "$host_installed_version" eq "$host_upgrade_version";
}
```

Reply:
IMHO. This is not quite necessary at this time. There is not too much occurrence of this code. I will take your suggestion into consideration in the future when it is necessary. OK ?

Reply:
Em......Yes, you are right.....If you insist on doing this, I will start doing so soon.

Reply:
@waynechen55 , I think this is a very valid suggestion from asmorodskyi, IMO, an excellent code should to reduce its redundancy.  Of course, you can give reason to reject.

Reply:
I will make some change.

Reply:
@waynechen55 @asmorodskyi  How about the PR rework? Ready to be reviewed again?